packages/apollo-engine-reporting/src/extension.ts

```diff
@@ -38,6 +38,9 @@ export class EngineReportingExtension<TContext = any>
   private options: EngineReportingOptions<TContext>;
   private addTrace: (args: AddTraceArgs) => Promise<void>;
   private generateClientInfo: GenerateClientInfo<TContext>;
+  private shouldReportQuery: (
+    request: GraphQLRequestContext<TContext>,
+  ) => boolean;
```

Comment:
Related to a comment below, how about a `Promise` of a `boolean` to allow `async` behavior that needs to be `await`ed within `shouldReportQuery`?:

```suggestion
  ) => Promise<boolean>;
```